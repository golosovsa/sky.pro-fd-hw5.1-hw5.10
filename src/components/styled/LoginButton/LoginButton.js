import styled from "styled-components";
import {
    layoutSizeMixin,
    layoutTextMixin,
    calcWidth,
    calcHeight,
} from "../../../globals/layout/index";
import { applyThemeMixin } from "../../../globals/theme/index";

const LoginButton = styled.button`
    ${layoutSizeMixin(278, 52)}
    ${layoutTextMixin(18, 24, 400)}
    letter-spacing: -0.003em;
    font-feature-settings: "pnum" on, "lnum" on;
    border-width: min(${calcWidth(1)}, ${calcHeight(1)});
    border-style: solid;
    border-radius: min(${calcWidth(6)}, ${calcHeight(6)});
    transition: background-color 0.25s ease-in-out;
    outline: none;
    &:hover,
    &:active,
    &:focus {
        outline: none;
        border-style: solid;
    }
    ${(props) => {
        if (props.type === "submit") {
            return applyThemeMixin("login", "submit");
        }
        return applyThemeMixin("login", "button");
    }}
`;

export default LoginButton;
