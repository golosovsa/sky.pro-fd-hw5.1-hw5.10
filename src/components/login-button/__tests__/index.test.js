import ShallowRenderer from 'react-test-renderer/shallow'
import LoginButton from '..'

const LoginButtonWrapped = ({ children }) => {
  return <LoginButton>{children}</LoginButton>
}

describe('<LoginButton />', () => {
  describe('Snapshot test', () => {
    it('should render successfully and match snapshot', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(<LoginButtonWrapped>123</LoginButtonWrapped>)

      expect(tree).toMatchSnapshot()
    })
  })
})
