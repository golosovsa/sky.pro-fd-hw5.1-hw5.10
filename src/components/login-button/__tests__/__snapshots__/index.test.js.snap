// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> Snapshot test should render successfully and match snapshot 1`] = `
<styled.input>
  123
</styled.input>
`;
